src/stan_math_backend/Expression_gen.ml

```diff
@@ -135,17 +135,15 @@ let fn_renames =
 let map_rect_calls = Int.Table.create ()
 let functor_suffix = "_functor__"
 let reduce_sum_functor_suffix = "_rsfunctor__"
-let variadic_ode_functor_suffix = "_odefunctor__"
-let variadic_dae_functor_suffix = "_daefunctor__"
+let variadic_functor_suffix x = sprintf "_vari_%d_functor__" x
```

Comment:
`_vari_` looks too much like it has something to do with `stan::math::vari` type.
```suggestion
let variadic_functor_suffix x = sprintf "_variadic%d_functor__" x
```
But also, I question whether these structs are needed. Per #583 it may be possible to replace these structs with C++ lambdas. `reduce_sum` and `map_rect` are handled separately anyway.
Or, if we're going to generate functors for all variadic HOFs then a separate `_rsfunctor__` isn't necessary, it can just be `_variadic4_functor__`.

Reply:
Agreed on the rename. 

I would like to eventually look into the lambda version (preferably after #1233) so for now I'm going to leave `_rsfunctor_`